### DISCUSSION TITLE
How I can log/audit the search/get/find methods?

### DISCUSSION THREAD

**Discussion by zarpelon** [spatie/laravel-activitylog#829] [Q&A]
Hey guys, Is It possible log the events like search/find/paginate on the package?

I try to override the $recordsEvents but don't works fine.

Or have a way to track by the route?

**Comment by Gummibeer** [comment]
Hey,
we can't hook into methods instead we can only use events fired by eloquent.
The matching event for these methods would be `retrieved` which you can monitor with this package.
I'm not sure about the `search` method. But you will lose which methods were used to retrieve the model.

The other solution would be custom activity logs. You could override the methods to add the log or add the log to your controllers or wherever you retrieve the models.
Alternatively you could use the repository pattern or something similar to add the log logic there.
